wordpress.org/public_html/wp-content/plugins/wporg-profiles-wp-activity-notifier/wporg-profiles-wp-activity-notifier.php

```diff
@@ -130,30 +120,84 @@ public function is_post_notifiable( $post ) {
 	/**
 	 * Only send notification for post getting published.
 	 *
-	 * @param string  $new_status The new status for the post
-	 * @param string  $old_status The old status for the post
-	 * @param WP_Post $post The post
+	 * @param string  $new_status The new status for the post.
+	 * @param string  $old_status The old status for the post.
+	 * @param WP_Post $post The post.
 	 */
 	public function maybe_notify_new_published_post( $new_status, $old_status, $post ) {
-		if ( 'publish' != $new_status ) {
+		if ( ! $this->is_post_notifiable( $post ) ) {
 			return;
 		}
 
-		if ( $old_status == $new_status ) {
-			return;
+		if ( 'wporg_workshop' === $post->post_type ) {
+			$this->notify_workshop_presenter( $post );
 		}
 
-		if ( ! $this->is_post_notifiable( $post ) ) {
+		if ( $old_status === $new_status ) {
 			return;
 		}
 
 		$this->notify_new_blog_post( $post );
 	}
 
+	/**
+	 * Sends activity notification for workshop presenter.
+	 *
+	 * @param WP_Post $post Post object.
+	 */
+	private function notify_workshop_presenter( $post ) {
+		if ( defined( 'WP_IMPORTING' ) && WP_IMPORTING ) {
+			return;
+		}
+
+		$presenter_wporg_username = filter_input( INPUT_POST, 'presenter-wporg-username' );
+
+		if ( empty( $presenter_wporg_username ) ) {
+			return;
+		}
+
+		$unique_presenter_wporg_username = array_unique( array_map( 'trim', explode( ',', $presenter_wporg_username ) ) );
+
+		$permalink = get_permalink( $post );
+
+		$title = wp_kses_data( $post->post_title );
+
+		$content = wp_trim_words(
+			strip_shortcodes( has_excerpt( $post ) ? $post->post_excerpt : $post->post_content ),
+			55
+		);
+
+		foreach ( $unique_presenter_wporg_username as $username ) {
+			$user_id = get_user_by( 'slug', strtolower( $username ) )->ID;
+
+			if ( ! $user_id ) {
+				continue;
+			}
+
+			$args = array(
+				'action'       => 'wporg_handle_activity',
+				'component'    => 'learn',
+				'type'         => 'workshop_presenter_assign',
+				'user_id'      => $user_id,
+				'primary_link' => $permalink,
+				'item_id'      => $post->ID,
+				'content'      => $content,
+				'message'      => sprintf(
+					'Assigned presenter to the tutorial, <i><a href="%s">%s</a></i>, on the site %s',
+					$permalink,
+					$title,
+					get_bloginfo( 'name' )
```

Comment:
```suggestion
					'Assigned as a presenter on the Learn WordPress tutorial, <i><a href="%s">%s</a></i>',
					$permalink,
					$title
```

This feels like it flows a bit smoother to me, but I don't have a strong opinion.

Reply:
👍 Yours is better. Changed.